tests/API/Management/JobsTest.php

```diff
@@ -0,0 +1,228 @@
+
+    /**
+     * Expected telemetry value.
+     *
+     * @var string
+     */
+    protected static $expectedTelemetry;
+
+    /**
+     * Default request headers.
+     *
+     * @var array
+     */
+    protected static $headers = [ 'content-type' => 'json' ];
+
+    /**
+     * Runs before test suite starts.
+     */
+    public static function setUpBeforeClass()
+    {
+        $infoHeadersData = new InformationHeaders;
+        $infoHeadersData->setCorePackage();
+        self::$expectedTelemetry = $infoHeadersData->build();
+    }
+
+    /**
+     * @throws \Exception Should not be thrown in this test.
+     */
+    public function testThatGetRequestIsFormedProperly()
+    {
+        $api = new MockManagementApi( [ new Response( 200, self::$headers ) ] );
+
+        $api->call()->jobs->get( '__test_id__' );
+
+        $this->assertEquals( 'GET', $api->getHistoryMethod() );
+        $this->assertEquals( 'https://api.test.local/api/v2/jobs/__test_id__', $api->getHistoryUrl() );
+
+        $headers = $api->getHistoryHeaders();
+        $this->assertEquals( 'Bearer __api_token__', $headers['Authorization'][0] );
+        $this->assertEquals( self::$expectedTelemetry, $headers['Auth0-Client'][0] );
+    }
+
+    /**
+     * @throws \Exception Should not be thrown in this test.
+     */
+    public function testThatGetErrorsIsFormedProperly()
+    {
+        $api = new MockManagementApi( [ new Response( 200, self::$headers ) ] );
+
+        $api->call()->jobs->getErrors( '__test_id__' );
+
+        $this->assertEquals( 'GET', $api->getHistoryMethod() );
+        $this->assertEquals( 'https://api.test.local/api/v2/jobs/__test_id__/errors', $api->getHistoryUrl() );
+
+        $headers = $api->getHistoryHeaders();
+        $this->assertEquals( 'Bearer __api_token__', $headers['Authorization'][0] );
+        $this->assertEquals( self::$expectedTelemetry, $headers['Auth0-Client'][0] );
+    }
+
+    /**
+     * @throws \Exception Should not be thrown in this test.
+     */
+    public function testThatImportUsersRequestIsFormedProperly()
+    {
+        $api = new MockManagementApi( [ new Response( 200, self::$headers ) ] );
+
+        $api->call()->jobs->importUsers(
+            self::TEST_IMPORT_USERS_JSON_PATH,
+            '__test_conn_id__',
+            [
+                'upsert' => true,
+                'send_completion_email' => true,
+                'external_id' => '__test_ext_id__',
+            ]
+        );
+
+        $this->assertEquals( 'POST', $api->getHistoryMethod() );
+        $this->assertEquals( 'https://api.test.local/api/v2/jobs/users-imports', $api->getHistoryUrl() );
+
+        $headers = $api->getHistoryHeaders();
+        $this->assertEquals( 'Bearer __api_token__', $headers['Authorization'][0] );
+        $this->assertEquals( self::$expectedTelemetry, $headers['Auth0-Client'][0] );
+        $this->assertStringStartsWith( 'multipart/form-data', $headers['Content-Type'][0] );
```

Comment:
why starts with vs equals? also, curious on why this request is using form vs body?

Reply:
> why starts with vs equals?

The URL includes the query parameters and I just want the URL here. 

> curious on why this request is using form vs body?

Import users endpoint requires a file to be posted.